src/Porpaginas/Twig/PorpaginasExtension.php

```diff
@@ -26,32 +26,22 @@ class PorpaginasExtension extends Twig_Extension
      */
     private $adapter;
 
-    /**
-     * @var \Twig_Environment
-     */
-    private $environment;
-
     public function __construct(RenderingAdapter $adapter)
     {
         $this->adapter = $adapter;
     }
 
-    public function initRuntime(Twig_Environment $environment)
-    {
-        $this->environment = $environment;
-    }
-
     public function getFunctions()
     {
         return array(
-            new Twig_SimpleFunction('porpaginas_render', array($this, 'renderPagination'), array('is_safe' => array('html'))),
+            new Twig_SimpleFunction('porpaginas_render', array($this, 'renderPagination'), array('is_safe' => array('html'), 'needs_environment' => true)),
             new Twig_SimpleFunction('porpaginas_total', array($this, 'renderTotal')),
         );
     }
 
-    public function renderPagination(Page $page)
+    public function renderPagination(Twig_Environment $environment, Page $page)
```

Comment:
since Twig 1.0.0-RC1. This was implemented when functions became first-class citizens in Twig: https://github.com/twigphp/Twig/commit/c225a5bed92616e69b13888e47ca2f99fe26731e